pkg/kn/commands/revision/describe.go

```diff
@@ -46,19 +62,225 @@ func NewRevisionDescribeCommand(p *commands.KnParams) *cobra.Command {
+func WriteImage(dw printers.PrefixWriter, revision *v1alpha1.Revision) {
+	c, err := clientserving.ContainerOfRevisionSpec(&revision.Spec)
+	if err != nil {
+		dw.WriteAttribute("Image", "Unknown")
+		return
+	}
+	image := c.Image
+	// Check if the user image is likely a more user-friendly description
+	pinnedDesc := "at"
+	userImage := clientserving.UserImage(&revision.ObjectMeta)
+	imageDigest := revision.Status.ImageDigest
+	if userImage != "" && imageDigest != "" {
+		var parts []string
+		if strings.Contains(image, "@") {
+			parts = strings.Split(image, "@")
+		} else {
+			parts = strings.Split(image, ":")
+		}
+		// Check if the user image refers to the same thing.
+		if strings.HasPrefix(userImage, parts[0]) {
+			pinnedDesc = "pinned to"
+			image = userImage
+		}
+	}
+	if imageDigest != "" {
+		image = fmt.Sprintf("%s (%s %s)", image, pinnedDesc, shortenDigest(imageDigest))
+	}
+	dw.WriteAttribute("Image", image)
+}
+
+func WritePort(dw printers.PrefixWriter, revision *v1alpha1.Revision) {
+	port := clientserving.Port(&revision.Spec)
+	if port != nil {
+		dw.WriteAttribute("Port", strconv.FormatInt(int64(*port), 10))
+	}
+}
+
+func WriteEnv(dw printers.PrefixWriter, revision *v1alpha1.Revision, printDetails bool) {
+	env := stringifyEnv(revision)
+	if env != nil {
+		commands.WriteSliceDesc(dw, env, "Env", printDetails)
+	}
+}
+
+func WriteScale(dw printers.PrefixWriter, revision *v1alpha1.Revision) {
+	// Scale spec if given
+	scale, _ := clientserving.ScalingInfo(&revision.ObjectMeta)
+	if scale != nil && (scale.Max != nil || scale.Min != nil) {
+		dw.WriteAttribute("Scale", formatScale(scale.Min, scale.Max))
+	}
+}
+
+func WriteResources(dw printers.PrefixWriter, r *v1alpha1.Revision) {
+	c, err := clientserving.ContainerOfRevisionSpec(&r.Spec)
+	if err != nil {
+		return
+	}
+	requests := c.Resources.Requests
+	limits := c.Resources.Limits
+	writeResourcesHelper(dw, "Memory", requests.Memory(), limits.Memory())
+	writeResourcesHelper(dw, "CPU", requests.Cpu(), limits.Cpu())
+}
+
+// Write request ... limits or only one of them
+func writeResourcesHelper(dw printers.PrefixWriter, label string, request *resource.Quantity, limit *resource.Quantity) {
+	value := ""
+	if !request.IsZero() && !limit.IsZero() {
+		value = request.String() + " ... " + limit.String()
+	} else if !request.IsZero() {
+		value = request.String()
+	} else if !limit.IsZero() {
+		value = limit.String()
+	}
+
+	if value == "" {
+		return
+	}
+
+	dw.WriteAttribute(label, value)
+}
+
+// Extract pure sha sum and shorten to 8 digits,
+// as the digest should to be user consumable. Use the resource via `kn service get`
+// to get to the full sha
+func shortenDigest(digest string) string {
+	match := imageDigestRegexp.FindStringSubmatch(digest)
+	if len(match) > 1 {
+		return string(match[1][:6])
+	}
+	return digest
+}
+
+// Format scale in the format "min ... max" with max = ∞ if not set
+func formatScale(minScale *int, maxScale *int) string {
```

Comment:
Should we move that up to a `FormatScale`, reusable by `service describe` ?

Reply:
Oh dang, it looks like I slightly misued git and forgot part of the refactor when I resurrected this PR. Resurrecting that :)